client/components/external-link/with-tracking.jsx

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import ExternalLink from './index.jsx';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+const ExternalLinkWIthTracking = ( {
+	onClick,
+	recordTracksEvent: recordEvent,
+	tracksEventName,
+	tracksEventProps,
+	...props
+} ) => {
+	const clickHandler = () => {
```

Comment:
I'm not super happy with this being defined on every re-render. Let's make it a method once we convert this to a class.